File: server/dto/match_action_dto.py
from dataclasses import dataclass

from dto.base_dto import BaseDto
from dto.coordinates_dto import CoordinatesDto
from dto.partial_spell_dto import PartialSpellDto
from game_engine.models.actions.action_type import ActionType


@dataclass
class MatchActionDto(BaseDto):
    player1: bool
    type: ActionType
    originatingCellCoords: CoordinatesDto
    impactedCoords: list[CoordinatesDto]
    spell: PartialSpellDto | None

File: client/src/dto/MatchActionDto.ts
import { ActionType } from "../enums/actionType";
import { CoordinatesDto } from "./CoordinatesDto";
import { PartialSpellDto } from "./PartialSpellDto";

export interface MatchActionDto {
    player1: boolean;
    type: ActionType;
    // Spell castings have no originating coordinates
    originatingCellCoords: CoordinatesDto | null;
    impactedCoords: CoordinatesDto[];
    // Only if the action type is spell
    spell: PartialSpellDto | null;
}